packages/manager/apps/dedicated/client/app/billing/autoRenew/autorenew.service.js

```diff
@@ -38,17 +39,28 @@ export default class {
     this.DucUserContractService = DucUserContractService;
     this.OvhApiBillingAutorenewServices = OvhApiBillingAutorenewServices;
     this.OvhApiEmailExchange = OvhApiEmailExchange;
+    this.ovhFeatureFlipping = ovhFeatureFlipping;
     this.OvhHttp = OvhHttp;
     this.ovhPaymentMethod = ovhPaymentMethod;
     this.OvhApiMeAutorenew = OvhApiMeAutorenew;
+    this.isLegacy = $q.defer();
 
     this.events = {
       AUTORENEW_CHANGES: AUTORENEW_EVENT,
     };
+
+    this.checkIfLegacy();
   }
 
-  getAllServices() {
-    return this.OvhApiBillingAutorenewServices.Aapi().query().$promise;
+  checkIfLegacy() {
+    return this.ovhFeatureFlipping
+      .checkFeatureAvailability(['billing:billingServices'])
+      .then((availability) =>
+        this.isLegacy.resolve(
+          !availability.isFeatureAvailable('billing:billingServices'),
+        ),
+      )
+      .catch(() => this.isLegacy.resolve(true));
```

Comment:
Maybe we could have used a `componentProvider` and use one or the other depending on the type, that way cleaning up would have been easier and no need to necessarly keep all existing code